resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsPollingSnapshotSource.scala

```diff
@@ -24,12 +24,21 @@ import io.fabric8.kubernetes.client.KubernetesClient
 import scala.collection.JavaConverters._
 
 import org.apache.spark.SparkConf
+import org.apache.spark.annotation.{DeveloperApi, Since, Stable}
 import org.apache.spark.deploy.k8s.Config._
 import org.apache.spark.deploy.k8s.Constants._
 import org.apache.spark.internal.Logging
 import org.apache.spark.util.{ThreadUtils, Utils}
 
-private[spark] class ExecutorPodsPollingSnapshotSource(
```

Comment:
Hm, @dongjoon-hyun. But this has been hidden as `private[spark]` so far. Should we maybe just mark `@since` as `3.3.0`?

Reply:
Yes, that's true technically in general, @HyukjinKwon . For this one, it's a little awkward situation because this has been used for a long time to implement a new cluster manager having `ExternalClusterManager` interface which is also `private[spark]`.

After we make K8s GA at 3.1.1, we noticed that K8s eco-system is fragile because some new contributions easily broke the others because everything is `private`. That's the reason why I used the last signature change to show the historic version at 4e8701a77dff729c4e8e0ad39c16e2717c2c32fe, `SPARK-35280 Promote KubernetesUtils to DeveloperApi`. This is very specific situation in K8s domain. So, if you don't mind, I'd want to show the original version here.